docs/en/operations/system-tables.md

```diff
@@ -587,93 +587,152 @@ Columns:
 
 The `system.query_log` table registers two kinds of queries:
 
 1.  Initial queries that were run directly by the client.
 2.  Child queries that were initiated by other queries (for distributed query execution). For these types of queries, information about the parent queries is shown in the `initial_*` columns.
 
+Each query creates one or two rows in the `query_log` table, depending on the status of the query:
+
+1.  If the query execution was successful, two events with types 1 and 2 are created (see the `type` column).
+2.  If an error occurred during query processing, two events with types 1 and 4 are created.
+3.  If an error occurred before launching the query, a single event with type 3 is created.
+
 Columns:
 
--   `type` (`Enum8`) — Type of event that occurred when executing the query. Values:
+-   `type` ([Enum8](../sql-reference/data-types/enum.md)) — Type of an event that occurred when executing the query. Values:
     -   `'QueryStart' = 1` — Successful start of query execution.
     -   `'QueryFinish' = 2` — Successful end of query execution.
     -   `'ExceptionBeforeStart' = 3` — Exception before the start of query execution.
     -   `'ExceptionWhileProcessing' = 4` — Exception during the query execution.
--   `event_date` (Date) — Query starting date.
--   `event_time` (DateTime) — Query starting time.
--   `query_start_time` (DateTime) — Start time of query execution.
--   `query_duration_ms` (UInt64) — Duration of query execution.
--   `read_rows` (UInt64) — Number of read rows.
--   `read_bytes` (UInt64) — Number of read bytes.
--   `written_rows` (UInt64) — For `INSERT` queries, the number of written rows. For other queries, the column value is 0.
--   `written_bytes` (UInt64) — For `INSERT` queries, the number of written bytes. For other queries, the column value is 0.
--   `result_rows` (UInt64) — Number of rows in the result.
--   `result_bytes` (UInt64) — Number of bytes in the result.
--   `memory_usage` (UInt64) — Memory consumption by the query.
--   `query` (String) — Query string.
--   `exception` (String) — Exception message.
--   `stack_trace` (String) — Stack trace (a list of methods called before the error occurred). An empty string, if the query is completed successfully.
--   `is_initial_query` (UInt8) — Query type. Possible values:
+-   `event_date` ([Date](../sql-reference/data-types/date.md)) — Query starting date.
+-   `event_time` ([DateTime](../sql-reference/data-types/datetime.md)) — Query starting time.
+-   `query_start_time` ([DateTime](../sql-reference/data-types/datetime.md)) — Start time of query execution.
+-   `query_duration_ms` ([UInt64](../sql-reference/data-types/int-uint.md#uint-ranges)) — Duration of query execution.
+-   `read_rows` ([UInt64](../sql-reference/data-types/int-uint.md#uint-ranges)) — Number of read rows.
+-   `read_bytes` ([UInt64](../sql-reference/data-types/int-uint.md#uint-ranges)) — Number of read bytes.
```

Comment:
* read from where?
* needs some clarification on how various caches are accounted (page cached, uncompressed block cache)

Reply:
> read from where

It is a total number or rows read from all tables and table functions participated in query. It includes usual subqueries, subqueries for `IN` and `JOIN`. 
For remote tables it includes the total number of rows read in all replicas. It is not just the number of rows received form each replica - each replica sends it's own `read_rows` and `read_bytes` value, they are summed up.

Reply:
Caches does not affect this numbers.